core/src/main/scala/org/apache/spark/deploy/master/ui/ApplicationPage.scala

```diff
@@ -70,6 +70,16 @@ private[ui] class ApplicationPage(parent: MasterWebUI) extends WebUIPage("app")
             }
             </li>
             <li>
+              <span data-toggle="tooltip" title={ToolTips.APPLICATION_EXECUTOR_LIMIT}
+                    data-placement="right">
+                <strong>Executor Limit: </strong>
+                {
+                  if (app.executorLimit == Int.MaxValue) "Unlimited" else app.executorLimit
```

Comment:
Agreed. Actually, there is another confusing difference between executor limit and maxCores: `maxCores` is fixed for an application while `executorLimit` is mutable. I'm not sure how to clarify them just on UI.